### ISSUE TITLE
[FEATURE] Integrate govulncheck within the build process

### ISSUE THREAD

**Issue by mdryankov** [rokwire/groups-building-block#319]
The govulncheck is an attempt of adding vulnerability check as an additional safety measure. 
The tool will be used on daily basis when a new version is going to be released and propagated to the environments. 

Additonal change:
Updated Dockerfile.

**Comment by shurwit** [comment]
Hi @mdryankov, I have update the makefile in the Building Block template to make some improvements including adding `govulncheck` into the build process: https://github.com/rokwire/building-block-template-go/blob/develop/Makefile. Feel free to take a look and replace the makefile in this repo with the contents there. 

Please note that our build process will be blocked by vulnerabilities and I found that I needed to upgrade to the latest version of go (v1.19.5) in order for govulncheck to pass. In general I think it is a good idea for us to stay up to date with the latest version of go and all dependencies, especially when known vulnerabilities are reported, but this will make it a requirement. 

If you do apply this change, please also add the GitHub action here: https://github.com/rokwire/building-block-template-go/blob/develop/.github/workflows/build.yaml to this repo as well and make it a required check in the branch protection rules to ensure we do not merge changes to `develop` or `main` that don't build successfully.

Let me know if you have any questions or concerns. Thanks!

**Comment by mdryankov** [comment]
Many thanks @shurwit! This is beutiful! 

I suspended for a while, because didn't succeeded with the Makefile jorney. 

About the go version. I totally aggre with you. Keeping older versions (that includes the libraries) produce risk for supply chain attacks & zero days. So, yep we need to move forward such updates on regular basis

**Comment by mdryankov** [comment]
Hi @shurwit. I have updated the project. Due to the minority of codechanges will postpone the release with a feature or two. 

Many thanks for your help!

CC @petyo @stefanvit
